lang/cs/Org.Apache.REEF.Wake/Remote/Impl/RemoteEventStreamingCodec.cs

```diff
@@ -30,65 +31,33 @@ namespace Org.Apache.REEF.Wake.Remote.Impl
     /// Writable remote event class
     /// </summary>
     /// <typeparam name="T">Type of remote event message. It is assumed that T implements IWritable</typeparam>
-    [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
-    internal sealed class WritableRemoteEvent<T> : IWritableRemoteEvent<T> where T : IWritable
+    internal sealed class RemoteEventStreamingCodec<T> : IStreamingCodec<IRemoteEvent<T>>
     {
-        private readonly IInjector _injector;
+        private readonly IStreamingCodec<T> _codec;
 
-        /// <summary>
-        /// Creates the Remote Event
-        /// </summary>
-        /// <param name="localEndpoint">Local Address</param>
-        /// <param name="remoteEndpoint">Remote Address</param>
-        /// <param name="value">Actual message</param>
-        public WritableRemoteEvent(IPEndPoint localEndpoint, IPEndPoint remoteEndpoint, T value)
+        public RemoteEventStreamingCodec(IStreamingCodec<T> codec)
         {
-            LocalEndPoint = localEndpoint;
-            RemoteEndPoint = remoteEndpoint;
-            Value = value;
-        }
-
-        /// <summary>
-        /// Creates empty Remote Event
-        /// </summary>
-        [Inject]
-        public WritableRemoteEvent(IInjector injector)
-        {
-            _injector = injector;
-        }
-
-        /// <summary>
-        /// Local Address
-        /// </summary>
-        public IPEndPoint LocalEndPoint { get; set; }
-
-        /// <summary>
-        /// Remote Address
-        /// </summary>
-        public IPEndPoint RemoteEndPoint { get; set; }
-
-        /// <summary>
-        /// The actual message
-        /// </summary>
-        public T Value { get; set; }
-
+            _codec = codec;
+        } 
+        
         /// <summary>
         /// Read the class fields.
         /// </summary>
         /// <param name="reader">The reader from which to read </param>
-        public void Read(IDataReader reader)
+        /// <returns>The remote event</returns>
+        public IRemoteEvent<T> Read(IDataReader reader)
         {
-            Value = (T)_injector.ForkInjector().GetInstance(typeof(T));
-            Value.Read(reader);         
+            return new RemoteEvent<T>(null, null, _codec.Read(reader));
         }
 
         /// <summary>
         /// Writes the class fields.
         /// </summary>
+        /// <param name="value">The remote event</param>
         /// <param name="writer">The writer to which to write</param>
-        public void Write(IDataWriter writer)
+        public void Write(IRemoteEvent<T> value, IDataWriter writer)
         {
-            Value.Write(writer);           
+            _codec.Write(value.Value, writer);           
```

Comment:
The fields Source, Sink, Sequence are never used. In earlier WritableRemoteEvent I removed them since I created that class. In RemoteEvent they were already there. So did not want to remove them. I can do it though. Let me know. Regarding LocalEndPoint and RemoteEndPoint see my comment above.